spark/src/main/scala/org/apache/spark/sql/delta/files/TransactionalWrite.scala

```diff
@@ -308,7 +308,7 @@ trait TransactionalWrite extends DeltaLogging { self: OptimisticTransactionImpl
       statsCollection: StatisticsCollection): Expression = {
     Dataset.ofRows(spark, LocalRelation(statsDataSchema))
       .select(to_json(statsCollection.statsCollector))
-      .queryExecution.analyzed.expressions.head
+      .queryExecution.optimizedPlan.expressions.head
```

Comment:
Needed because of https://github.com/apache/spark/commit/b4eb03469a90ec5bf65fd51d2bcb41602acad51b